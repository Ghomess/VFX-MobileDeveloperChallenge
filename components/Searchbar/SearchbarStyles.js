import {StyleSheet} from 'react-native';
import {colors} from '../../styles';

export const searchbarStyles = StyleSheet.create({
  container: {
    width: '90%',
    flexDirection: 'row',
    paddingHorizontal: 10,
    alignItems: 'center',
    backgroundColor: colors.blue,
    borderTopRightRadius: 5,
    borderTopLeftRadius: 5,
    borderBottomColor: colors.darkgray,
    borderBottomWidth: 2,
  },
  textInput: {
    width: '88%',
    fontSize: 15,
  },
});
