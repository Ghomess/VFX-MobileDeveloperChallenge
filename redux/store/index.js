import {configureStore} from '@reduxjs/toolkit';
import stockSlice from '../reducers/stockSlice';
import pairSlice from '../reducers/pairSlice';
import loadingSlice from '../reducers/loadingSlice';

const store = configureStore({
  reducer: {
    ticker: stockSlice,
    pair: pairSlice,
    loading: loadingSlice,
  },
});

export default store;
