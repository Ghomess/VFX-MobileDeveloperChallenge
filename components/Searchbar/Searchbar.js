import React from 'react';
import {TextInput, View} from 'react-native';
import {searchbarStyles} from './SearchbarStyles';
import IonicIcons from 'react-native-vector-icons/Ionicons';

export const Searchbar = ({value, setValue, placeHolder, setIsFocused}) => {
  return (
    <View style={searchbarStyles.container}>
      <IonicIcons name="search" color={'#ffffff'} size={20} />
      <TextInput
        style={searchbarStyles.textInput}
        value={value}
        onChangeText={setValue}
        placeholder={placeHolder}
        placeholderTextColor={'#8C8C8C'}
        onFocus={() => setIsFocused(true)}
        onBlur={() => setIsFocused(false)}
        color={'#ffffff'}
      />
    </View>
  );
};
