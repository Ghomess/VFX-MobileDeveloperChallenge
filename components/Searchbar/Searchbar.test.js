import React, {useState} from 'react';

import {NavigationContainer} from '@react-navigation/native';
import {Provider} from 'react-redux';
import store from '../../redux/store';
import {fireEvent, render, waitFor} from '@testing-library/react-native';
import {Searchbar} from './Searchbar';

jest.mock('@react-navigation/native', () => ({
  ...jest.requireActual('@react-navigation/native'),
  useNavigation: () => ({
    getState: () => ({
      routes: [{name: 'Stocks'}],
      index: 0,
    }),
  }),
}));

test('renders correctly', async () => {
  await waitFor(() => {
    const tree = render(
      <Provider store={store}>
        <NavigationContainer>
          <Searchbar
            placeHolder={'Search ticker'}
            setValue={jest.fn()}
            value={jest.fn()}
            setIsFocused={jest.fn()}
          />
        </NavigationContainer>
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
});

//More inside Search.test.js
