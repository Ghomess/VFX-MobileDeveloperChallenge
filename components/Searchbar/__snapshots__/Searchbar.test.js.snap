// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#1A237E",
      "borderBottomColor": "#424242",
      "borderBottomWidth": 2,
      "borderTopLeftRadius": 5,
      "borderTopRightRadius": 5,
      "flexDirection": "row",
      "paddingHorizontal": 10,
      "width": "90%",
    }
  }
>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      [
        {
          "color": "#ffffff",
          "fontSize": 20,
        },
        undefined,
        {
          "fontFamily": "Ionicons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        {},
      ]
    }
  >
    î¹ 
  </Text>
  <TextInput
    color="#ffffff"
    onBlur={[Function]}
    onChangeText={[MockFunction]}
    onFocus={[Function]}
    placeholder="Search ticker"
    placeholderTextColor="#8C8C8C"
    style={
      {
        "fontSize": 15,
        "width": "88%",
      }
    }
    value={[MockFunction]}
  />
</View>
`;
